src/main/java/org/folio/rest/impl/InitAPIs.java

```diff
@@ -78,21 +75,47 @@ public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> res
+    SpringContextUtil.autowireDependencies(this, context);
+    log.info("initSpringContext:: Spring context was successfully started");
+  }
+
+  private void initKafkaConsumersVerticles(Vertx vertx, Promise<Object> handler) {
+    deployKafkaConsumersVerticles(vertx).onComplete(ar -> {
+      if (ar.failed() && isConsumersVerticleMandatory) {
+        handler.fail(ar.cause());
+      } else {
+        handler.complete();
+      }
+    });
+  }
+
   private Future<?> deployKafkaConsumersVerticles(Vertx vertx) {
-    AbstractApplicationContext springContext = vertx.getOrCreateContext().get("springContext");
+    var springContext = (AbstractApplicationContext) vertx.getOrCreateContext().get(SPRING_CONTEXT_KEY);
     var consumers = List.of(
-      deployVerticle("inventory-item-consumers", itemConsumersVerticleNumber, InventoryItemConsumersVerticle.class, vertx, springContext),
-      deployVerticle("inventory-holding-consumers", holdingConsumersVerticleNumber, InventoryHoldingConsumersVerticle.class, vertx, springContext),
-      deployVerticle("edi-export-orders-history-consumers", ediExportConsumersVerticleNumber, EdiExportOrdersHistoryConsumersVerticle.class, vertx, springContext)
+      deployVerticle("inventory-item-consumers", itemConsumerVerticleNumber, itemConsumerPoolSize, InventoryItemConsumersVerticle.class, vertx, springContext),
+      deployVerticle("inventory-holding-create-consumers", holdingCreateConsumerVerticleNumber, holdingCreateConsumerPoolSize, InventoryHoldingCreateConsumersVerticle.class, vertx, springContext),
+      deployVerticle("inventory-holding-update-consumers", holdingUpdateConsumerVerticleNumber, holdingUpdateConsumerPoolSize, InventoryHoldingUpdateConsumersVerticle.class, vertx, springContext),
+      deployVerticle("edi-export-orders-history-consumers", ediExportConsumerVerticleNumber, ediExportConsumerPoolSize, EdiExportOrdersHistoryConsumersVerticle.class, vertx, springContext)
     );
     return GenericCompositeFuture.all(consumers)
       .onSuccess(future -> log.info("deployKafkaConsumersVerticles:: All {} consumer(s) was successfully started", consumers.size()))
       .onFailure(exception -> log.error("Failed to start consumers", exception));
   }
 
-  private <V extends Verticle> Future<String> deployVerticle(String poolName, int instancesNumber, Class<V> consumerClass, Vertx vertx, AbstractApplicationContext springContext) {
-    return vertx.deployVerticle(() -> springContext.getBean(consumerClass),
-      new DeploymentOptions().setThreadingModel(ThreadingModel.WORKER).setWorkerPoolName(poolName).setInstances(instancesNumber));
+  private <V extends Verticle> Future<String> deployVerticle(String poolName, int instancesNumber, int poolSize, Class<V> consumerClass, Vertx vertx, AbstractApplicationContext springContext) {
+    var deploymentOptions = new DeploymentOptions().setThreadingModel(ThreadingModel.WORKER).setWorkerPoolName(poolName).setInstances(instancesNumber).setWorkerPoolSize(poolSize);
```

Comment:
seems some lines is too long can be moved to next line. please look at other places as well. long lines is hard to read in laptop screen or dividied screen in monitor